From: Mike <mikeandmore@gmail.com>
Description: force some flags always on in linkflags

 * SConstruct: force some flags always on in linkflags

Origin: upstream

--- a/SConstruct
+++ b/SConstruct
@@ -180,12 +180,6 @@
 libdatadir = env['LIBDATADIR'] + '/sunpinyin/data'
 headersdir = env['PREFIX'] + '/include/sunpinyin-2.0'
 
-if GetOS() != 'Darwin':
-    env.Append(LINKFLAGS=['-Wl,-soname=libsunpinyin.so.%d' % abi_major])
-
-if GetOption('rpath') is not None and GetOS() != 'Darwin':
-    env.Append(LINKFLAGS='-Wl,-R -Wl,%s' % GetOption('rpath'))
-
 # pass through environmental variables
 envvar = [('CC', 'CC'),
           ('CXX', 'CXX'),
@@ -202,6 +196,12 @@
 env.Append(CFLAGS=extra_cflags)
 env.Append(CXXFLAGS=extra_cflags)
 
+if GetOS() != 'Darwin':
+    env.Append(LINKFLAGS=['-Wl,-soname=libsunpinyin.so.%d' % abi_major])
+
+if GetOption('rpath') is not None and GetOS() != 'Darwin':
+    env.Append(LINKFLAGS=['-Wl,-R', '-Wl,%s' % GetOption('rpath')])
+
 #
 #==============================configure================================
 #
